File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p04034/p04034_num0_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N, M;
    cin >> N >> M;

    vector<int> boxes(N, 1); // Initialize all boxes with white balls
    boxes[0] = 2; // Setting box 1 with a red ball

    for (int i = 0; i < M; i++) {
        int x, y;
        cin >> x >> y;

        boxes[x-1]--; // Remove a ball from box x
        boxes[y-1]++; // Add a ball to box y
        if (boxes[x-1] == 0) {
            boxes[x-1] = -1; // Mark box x as empty
        }
    }

    int count = 0;
    for (int i = 0; i < N; i++) {
        if (boxes[i] > 0) {
            count++;
        }
    }

    cout << count << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p04034/p04034_num0_parsed.py
# Program

# Read N and M
N, M = map(int, input().split())

# Initialize a list to keep track of the number of balls in each box
balls = [1] + [0] * (N - 1)

# Iterate through each operation
for _ in range(M):
    # Read the operation
    x, y = map(int, input().split())

    # Decrement the number of balls in box x
    balls[x - 1] -= 1

    # Increment the number of balls in box y
    balls[y - 1] += 1

# Count the number of boxes that may contain the red ball
count = sum(1 for ball in balls if ball > 0)

# Print the result
print(count)